runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -73,18 +73,24 @@ public Object getPayload(CorfuRuntime runtime) {
                     if (data == null) {
                         this.payload.set(null);
                     } else {
-                        ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
-                        final Object actualValue =
-                                Serializers.CORFU.deserialize(copyBuf, runtime);
-                        if (actualValue instanceof LogEntry) {
-                            ((LogEntry) actualValue).setGlobalAddress(getGlobalAddress());
-                            ((LogEntry) actualValue).setRuntime(runtime);
+                        try {
+                            ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
+                            final Object actualValue =
+                                    Serializers.CORFU.deserialize(copyBuf, runtime);
```

Comment:
Limit the scope of the try/catch to the deserialize line ?

Reply:
done.